src/lib/es2015.symbol.wellknown.d.ts

```diff
@@ -219,9 +219,9 @@ interface String {
     match(matcher: { [Symbol.match](string: string): RegExpMatchArray | null; }): RegExpMatchArray | null;
 
     /**
-     * Replaces first match with string or all matches with RegExp.
-     * @param searchValue A string or RegExp search value.
-     * @param replaceValue A string containing the text to replace for match.
+     * Replaces one or more occurrences of substrings that match the method provided by `searchValue`.

```

Comment:
I'm surprised that it will replace multiple instances, but not based on any actual experience.  You confirmed somehow?

Reply:
This is the doc for `@@replace` so it can have any custom behavior—only when `searchValue` is a string will it only replace once.

Reply:
The custom behaviour is the point, so I think it'd be better to talk about that. Maybe:

```suggestion
     * Passes a string and {@linkcode replaceValue} to the [Symbol.replace] method on {@linkCode searchValue}. This method is expected to implement its own replacement algorithm.
```

I'm not sure whether

1. `[Symbol.replace]` is enough explanation of how to construct a good searchValue.
2. It's actually true that the method is expected to implement its own replacement algorithm.

Reply:
Is it possible that we add another `replacementValue: RegExp` overload before this? FWIW, replacing with regexes remains what people write 99% of the time and it makes sense to have some doc more specific about that.

> It's actually true that the method is expected to implement its own replacement algorithm.

I believe `@@replace` was implemented purely to make string methods decoupled from regex semantics. Nevertheless, yes, it can have any custom behavior.